internal/entitlement/connector.go

```diff
@@ -33,68 +33,98 @@ type EntitlementConnector interface {
 }
 
 type entitlementConnector struct {
-	entitlementBalanceConnector EntitlementBalanceConnector
-	entitlementRepo             EntitlementRepo
-	featureConnector            productcatalog.FeatureConnector
+	meteredEntitlementConnector SubTypeConnector
+	staticEntitlementConnector  SubTypeConnector
+	booleanEntitlementConnector SubTypeConnector
+
+	entitlementRepo  EntitlementRepo
+	featureConnector productcatalog.FeatureConnector
 }
 
 func NewEntitlementConnector(
-	entitlementBalanceConnector EntitlementBalanceConnector,
 	entitlementRepo EntitlementRepo,
 	featureConnector productcatalog.FeatureConnector,
-) EntitlementConnector {
+	meteredEntitlementConnector SubTypeConnector,
```

Comment:
Most probably not for this PR, but this is getting out of hand, esp if we have to maintain this initialization sequence in multiple components. Can we have a single helper initialize the whole feature and return the connectors?

Reply:
will squeeze in in one of the followups